File: src/data/PlayerDataProvider.ts
import { Player } from "../types/hytale";

export type PlayerData = {
  rank: string;
  coins: number;
  level: number;
  gameName: string;
};

export class PlayerDataProvider {
  getPlayerData(player: Player): PlayerData {
    return {
      rank: player.rank ?? "Aventurier",
      coins: player.coins ?? 0,
      level: player.level ?? 1,
      gameName: "Mini-jeu",
    };
  }
}

File: src/main/java/com/hytale/scoreboard/data/PlayerDataProvider.java
package com.hytale.scoreboard.data;

import com.hytale.scoreboard.types.Player;

public class PlayerDataProvider {
  public PlayerData getPlayerData(Player player) {
    String rank = player.getRank() != null ? player.getRank() : "Aventurier";
    int coins = player.getCoins() != null ? player.getCoins() : 0;
    int level = player.getLevel() != null ? player.getLevel() : 1;

    return new PlayerData(rank, coins, level, "Mini-jeu");
  }
}
